R/tt_pos_and_access.R

```diff
@@ -327,52 +327,90 @@ setMethod("tt_at_path<-", c(tt = "VTableTree", value = "TableRow"),
+#' \item{\emph{keep_fnotes}}{logical(1) Should non-referential footer
+#' information be retained. Defaults to \code{keep_titles}.}
 #' \item{\emph{reindex_refs}}{logical(1). Should referential footnotes be
 #' re-indexed as if the resulting subset is the entire table. Defaults to
-#' \code{TRUE}}
+#' \code{TRUE}.}
 #' }
 #' @param value Replacement value (list, `TableRow`, or `TableTree`)
+#' 
 #' @return a \code{TableTree} (or \code{ElementaryTable}) object, unless a
 #'   single cell was selected with \code{drop=TRUE}, in which case the (possibly
 #'   multi-valued) fully stripped raw value of the selected cell.
-#' @exportMethod [<-
+#' 
+#' @details the standard subsetting drops the information about title, subtitle,
+#' main footer, provenance footer, and `topleft`. If only a column is selected and
+#' all rows are kept, the `topleft` information remains as default. Any referential
+#' footnote is kept whenever the subset table contains the referenced element. 
+#' Please note that subsetting always preserve the original order, even if provided
+#' indexes do not preserve it. If sorting is needed, please consider 
+#' using `sort_at_path()`. Also note that `character` indices are treated as paths,
+#' not vectors of names in both `[` and `[<-`.
+#' 
+#' @seealso Regarding sorting: `sort_at_path()` and how to understand path 
+#' structure: `summarize_row_groups()`, and `summarize_col_groups()`.
 #'
 #' @examples
-#' lyt <- basic_table() %>%
+#' lyt <- basic_table(title = "Title", 
+#'                    subtitles = c("Sub", "titles"), 
+#'                    prov_footer = "prov footer", 
+#'                    main_footer = "main footer") %>%
 #'    split_cols_by("ARM") %>%
-#'    analyze(c("SEX", "AGE"))
+#'    split_rows_by("SEX") %>%
+#'    analyze(c("AGE"))
 #'
 #' tbl <- build_table(lyt, DM)
+#' top_left(tbl) <- "Info"
 #' tbl
-#'
+#' 
+#' # As default header, footer, and topleft information is dropped
 #' tbl[1, ]
 #' tbl[1:2, 2]
-#'
+#' 
+#' # If drop = TRUE, the content values are directly retrieved
 #' tbl[2, 1]
 #' tbl[2, 1, drop = TRUE]
-#'
+#' 
+#' # If all rows are selected, topleft is kept by default
+#' tbl[, 2]
 #' tbl[, 1]
 #'
 #' tbl[-2, ]
 #' tbl[, -1]
 #'
 #' tbl[2, 1] <- rcell(999)
 #' tbl[2, ] <- list(rrow("FFF", 888, 666, 777))
-#' tbl[3, ] <- list(-111, -222, -333)
+#' tbl[6, ] <- list(-111, -222, -333)
 #' tbl
+#' 
+#' # We can keep some information from the original table if we need
+#' tbl[1, 2, keep_titles = TRUE]
+#' tbl[1, 2, keep_fnotes = TRUE, keep_titles = FALSE]
+#' tbl[1, 2, keep_fnotes = FALSE, keep_titles = TRUE]
+#' tbl[1, 2, keep_fnotes = TRUE]
+#' tbl[1, 2, keep_topleft = TRUE]
+#' 
+#' # Keeps the referential footnotes when subset contains them
+#' fnotes_at_path(tbl, rowpath = c("SEX", "M", "AGE", "Mean")) <- "important"
+#' tbl[4,1]
+#'
+#' # Note that order can not be changed with subsetting
+#' tbl[c(4,3,1), c(3,1)] # It preserves order and wanted selection
+#' 
```

Comment:
Add example using `reindex_refs= FALSE` and the default (`TRUE`) give different answers